### PR TITLE
Dev/test

### PR DESCRIPTION
**Pull request by wush978**
For https://github.com/r-dbi/RSQLite/issues/233#issuecomment-450576567

### CODE REVIEW

File: tests/testthat/test-encoding.R [hunk 1]
```diff
@@ -0,0 +1,160 @@
+context("encoding")
+
+test_that("write tables whose colnames or contents are BIG5 encoded", {
+  skip_if(.Platform$OS.type != "windows")
```
Comment (krlmlr) [inline tests/testthat/test-encoding.R]: There's also `skip_on_os()`.

File: tests/testthat/test-encoding.R [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  colnames(df) <- paste(., 1:2, sep = "")
+  dbWriteTable(con, "b", df)
+  res <- dbReadTable(con, "b")
+  expect_identical(res, df)
+})
+
+test_that("list the field of tables whose colnames are BIG5 encoded", {
+  skip_if(.Platform$OS.type != "windows")
+  .loc <- Sys.getlocale("LC_COLLATE")
+  if (.Platform$OS.type == "windows") {
```
Comment (krlmlr) [inline tests/testthat/test-encoding.R]: Do you need this check here?

File: tests/testthat/test-encoding.R [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  skip_if(.Platform$OS.type != "windows")
+  .loc <- Sys.getlocale("LC_COLLATE")
+  Sys.setlocale(locale = "cht")
+  con <- dbConnect(SQLite())
+  on.exit({
+    Sys.setlocale(locale = .loc)
+    dbDisconnect(con)
+  })
+
+  . <- rawToChar(as.raw(c(0xa4, 0xa4, 0xa4, 0xe5)))
```
Comment (krlmlr) [inline tests/testthat/test-encoding.R]: Could you please use a meaningful name here? The dot is used much further below, this causes confusion.

File: tests/testthat/test-encoding.R (hunk 2 above)
Comment (wush978) [reply tests/testthat/test-encoding.R]: Yes. As far as I know, the value returned by `make.names` is system dependent.

File: tests/testthat/test-encoding.R (hunk 3 above)
Comment (wush978) [reply tests/testthat/test-encoding.R]: Sure

**Comment by krlmlr** [comment]
Thanks!

**Comment by krlmlr** [comment]
The last test doesn't seem to include any expectations? What do we test here?

**Comment by wush978** [comment]
As I remember, they are testing `dbWriteTable` with `append = TRUE`. The tests will directly fail if the bug is not fixed.

**Pull request merged**